tests/llm/test_token_counter.py

```
@@ -0,0 +1,217 @@
+        """Should respect custom max_tokens parameter."""
+        prompt = "test" * 100  # ~100 tokens
+        # Custom limit of 50 tokens should raise
+        with pytest.raises(ContextLengthExceededError):
+            TokenCounter.validate_context_length(prompt, max_tokens=50)
+
+    def test_validate_context_length_exception_details(self):
+        """Exception should contain useful debugging info."""
+        prompt = "a" * 50000
+        try:
+            TokenCounter.validate_context_length(prompt, model_name='gpt-3.5-turbo')
+            assert False, "Should have raised"
```

Ruff (0.15.15)

[warning] Do not `assert False` (`python -O` removes these calls), raise `AssertionError()`

Replace `assert False`

(B011)